relay-dynamic-config/src/defaults.rs

```diff
@@ -106,6 +106,17 @@ pub fn add_span_metrics(project_config: &mut ProjectConfig) {
             ),
             tags: Default::default(),
```

Comment:
If no SDK is sending data with the wrong key, we can safely remove the old `MetricSpec`.